src/vs/workbench/contrib/remote/browser/explorerViewItems.ts

```diff
@@ -4,45 +4,56 @@
 import { IViewDescriptor } from 'vs/workbench/common/views';
 import { isStringArray } from 'vs/base/common/types';
 import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
 import { IStorageService, StorageScope } from 'vs/platform/storage/common/storage';
-import { ContextKeyExpr, IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
-import { SelectActionViewItem } from 'vs/base/browser/ui/actionbar/actionViewItems';
-import { Action2, MenuId } from 'vs/platform/actions/common/actions';
-import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
+import { ContextKeyExpr, IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { Action2, MenuId, MenuRegistry, registerAction2 } from 'vs/platform/actions/common/actions';
 import { VIEWLET_ID } from 'vs/workbench/contrib/remote/browser/remoteExplorer';
-import { defaultSelectBoxStyles } from 'vs/platform/theme/browser/defaultStyles';
 import { getVirtualWorkspaceLocation } from 'vs/platform/workspace/common/virtualWorkspace';
 import { IWorkspaceContextService } from 'vs/platform/workspace/common/workspace';
+import { Disposable } from 'vs/base/common/lifecycle';
 
 interface IRemoteSelectItem extends ISelectOptionItem {
 	authority: string[];
 	virtualWorkspace?: string;
 }
 
-export class SwitchRemoteViewItem extends SelectActionViewItem<IRemoteSelectItem> {
+export const SELECTED_REMOTE_IN_EXPLORER = new RawContextKey<string>('selectedRemoteInExplorer', '');
+
+export class SwitchRemoteViewItem extends Disposable {
+	private switchRemoteMenu: MenuId;
+	private options: IRemoteSelectItem[] = [];
+	private completedRemotes: Set<string> = new Set();
+	private readonly selectedRemoteContext: IContextKey<string>;
 
 	constructor(
-		action: IAction,
-		private readonly optionsItems: IRemoteSelectItem[],
-		@IContextViewService contextViewService: IContextViewService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
 		@IRemoteExplorerService private remoteExplorerService: IRemoteExplorerService,
 		@IWorkbenchEnvironmentService private environmentService: IWorkbenchEnvironmentService,
 		@IStorageService private readonly storageService: IStorageService,
 		@IWorkspaceContextService private readonly workspaceContextService: IWorkspaceContextService
 	) {
-		super(null, action, optionsItems, 0, contextViewService, defaultSelectBoxStyles, { ariaLabel: nls.localize('remotes', 'Switch Remote') });
+		super();
+		this.selectedRemoteContext = SELECTED_REMOTE_IN_EXPLORER.bindTo(contextKeyService);
+
+		this.switchRemoteMenu = new MenuId('workbench.remote.menu.switchRemoteMenu');
+		this._register(MenuRegistry.appendMenuItem(MenuId.ViewContainerTitle, {
+			submenu: this.switchRemoteMenu,
+			title: nls.localize('switchRemote.label', "Switch Remote"),
+			group: 'navigation',
+			when: ContextKeyExpr.equals('activeViewlet', VIEWLET_ID),
```

Comment:
```suggestion
			when: ContextKeyExpr.equals('viewContainer', VIEWLET_ID),
```

This should be `viewContainer` otherwise this action will be added to the view containers in panel and auxilary bar when the remote explorer is active.